docs/source/library-user-guide/working-with-exprs.md

```diff
@@ -19,4 +19,205 @@
 
 # Working with Exprs
 
-Coming Soon
+<!-- https://github.com/apache/arrow-datafusion/issues/7304 -->
+
+`Expr` is short for "expression". It is a core abstraction in DataFusion for representing a computation.
+
+For example, the SQL expression `a + b` would be represented as an `Expr` with a `BinaryExpr` variant. A `BinaryExpr` has a left and right `Expr` and an operator.
+
+As another example, the SQL expression `a + b * c` would be represented as an `Expr` with a `BinaryExpr` variant. The left `Expr` would be `a` and the right `Expr` would be another `BinaryExpr` with a left `Expr` of `b` and a right `Expr` of `c`.
```

Comment:
Eventually it would be cool to make a diagram to represent this. Maybe something like:

```text
            ┌────────────────────┐                                
            │     BinaryExpr     │                                
            │       op: +        │                                
            └────────────────────┘                                
                   ▲     ▲                                        
           ┌───────┘     └────────────────┐                       
           │                              │                       
┌────────────────────┐         ┌────────────────────┐             
│     Expr::Col      │         │     BinaryExpr     │             
│       col: a       │         │       op: *        │             
└────────────────────┘         └────────────────────┘             
                                        ▲    ▲                    
                               ┌────────┘    └─────────┐          
                               │                       │          
                    ┌────────────────────┐  ┌────────────────────┐
                    │     Expr::Col      │  │     Expr::Col      │
                    │       col: b       │  │       col: c       │
                    └────────────────────┘  └────────────────────┘
```